converter/internal/common/http_client_config.go

```diff
@@ -30,21 +30,10 @@ func ToHttpClientConfig(httpClientConfig *prom_config.HTTPClientConfig) *config.
 func ValidateHttpClientConfig(httpClientConfig *prom_config.HTTPClientConfig) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	if httpClientConfig.NoProxy != "" {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config no_proxy was provided")
-	}
-
-	if httpClientConfig.ProxyFromEnvironment {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config proxy_from_environment was provided")
-	}
-
-	if len(httpClientConfig.ProxyConnectHeader) > 0 {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config proxy_connect_header was provided")
-	}
-
-	if httpClientConfig.TLSConfig.MaxVersion != 0 {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config max_version was provided")
-	}
+	diags.AddAll(UnsupportedNotEquals(httpClientConfig.NoProxy, "", "HTTP Client no_proxy"))
```

Comment:
nit: Should this be InvalidNotEquals since we changed the wording?

Reply:
Hmmm, maybe not looking at it farther.

Reply:
I'm going to take a pass on these function names. They haven't kept up as the contents and location have evolved.